### ISSUE TITLE
improvement  on one_hot labels

### ISSUE THREAD

**Issue by zorrocai** [ICEORY/softmax_loss_gradient#1]
the loop in the generation the one hot label could be replaced by the scatter_() function.

```
t = torch.zeros(y.size()).type(y.type())
for n in range(t.size(0)):
      t[n][target[n]] = 1
```
could be replaced by the following
```
y_onehot = torch.zeros(y.size()).type(y.type())
 y_onehot.scatter_(1, target.unsqueeze(1), 1)

```
